FaunaDB.Client/Errors/NotNullValidator.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System;
+
+namespace FaunaDB.Errors
+{
+    static class NotNullValidator
+    {
+        public static void AssertNotNull(this object value, string valueName)
+        {
+            if (value == null)
+            {
+                throw new ArgumentNullException(valueName);
```

Comment:
can we use `nameof` here so the callers are even more simple?

```suggestion
                throw new ArgumentNullException(nameof(valueName));
```